### PR TITLE
an argument is not always needed to be passed to open a file

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Sometimes this is loaded using just the default, which is 'r', and 'r' is not passed.  we should still be able to open the file.

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by gtmanfred** [comment]
@saltstack/team-core this would be super nice to get into 2018.3.4, since it is pretty minor and only broken starting in 2018.3.3

Thanks,
Daniel

**Comment by gtmanfred** [comment]
Broken by https://github.com/saltstack/salt/commit/eaed033cdea35dfb6a19cbf134614b134228593d

@dmurphy18 can you take a look?

Thanks,
Daniel

File: salt/utils/files.py [hunk 1]
```diff
@@ -416,11 +416,13 @@ def flopen(*args, **kwargs):
     '''
     Shortcut for fopen with lock and context manager.
     '''
-    with fopen(*args, **kwargs) as f_handle:
+    filename = args.pop(0)
+    vars = 'wa'
```
Comment (s0undt3ch) [inline salt/utils/files.py]: Could we use something else instead of `vars`? it shadows a built-in.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/files.py (hunk 1 above)
Comment (gtmanfred) [reply salt/utils/files.py]: yeah

**Comment by gtmanfred** [comment]
ok, fixed.

**Review by s0undt3ch — APPROVED** [review]

**Review by thatch45 — APPROVED** [review]

**Comment by thatch45** [comment]
Are these test failures related? This is some very central code :)

**Comment by gtmanfred** [comment]
checking.

**Comment by gtmanfred** [comment]
Ok, test should be fixed.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**